### PR TITLE
eOGBlxNe Added SignOut Event and confirmed tests

### PR DESCRIPTION
**Pull request by j-maynard**
**What**
Allow users to log out of their self-service account securely and end the session.

**Why**
Users are likely to expect this feature and we should provide it for security purposes.

We have already made some progress as part of the authentication, therefore this ticket is about validating the logout functionality and adding any tests (if neccessary)

**What we did**
Created a new SignOut event to log when users sign out.  Added related tests.  Confirmed sign out process.

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class SessionsController < Devise::SessionsController
+  def new
+    super
+  end
```
Comment (Rosa-Fox) [inline app/controllers/sessions_controller.rb]: we can remove this method

File: spec/models/user_sign_out_event_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  let(:user_id) { SecureRandom.uuid }
+  let(:event) { UserSignOutEvent.create(user_id: user_id) }
+  let(:event_with_nil_user_id) {
+    UserInfo.current_user = nil
+    UserSignInEvent.create(user_id: nil)
+  }
+
+  context '#create' do
+    it 'a valid event which contains a user id' do
+
```
Comment (Rosa-Fox) [inline spec/models/user_sign_out_event_spec.rb]: We can remove this space... same with the sign_in spec from previous PR

File: spec/models/user_sign_out_event_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+require 'yaml'
```
Comment (Rosa-Fox) [inline spec/models/user_sign_out_event_spec.rb]: We don't need this as far as I can tell... same with the sign_in spec from previous PR

File: spec/models/user_sign_out_event_spec.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  let(:event) { UserSignOutEvent.create(user_id: user_id) }
+  let(:event_with_nil_user_id) {
+    UserInfo.current_user = nil
+    UserSignInEvent.create(user_id: nil)
+  }
+
+  context '#create' do
+    it 'a valid event which contains a user id' do
+
+      expect(event.user_id).to eq(user_id)
```
Comment (brenetic) [inline spec/models/user_sign_out_event_spec.rb]: Should we also check that it's a `UserSignOutEvent` type?

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+  def destroy
+    @logout_event = UserSignOutEvent.create(user_id: warden.user.user_id)
```
Comment (brenetic) [inline app/controllers/sessions_controller.rb]: Does Devise make use of the `@logout_event` variable?

**Pull request merged**